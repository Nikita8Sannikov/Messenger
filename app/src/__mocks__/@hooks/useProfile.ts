export default () => ({
    on: jest.fn(),
    off: jest.fn(),
    user: {
        avatarUrl: jest.fn(),
        fullName: jest.fn()
    },
})