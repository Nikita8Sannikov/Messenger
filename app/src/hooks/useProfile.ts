import useMainClient from "@hooks/useMainClient";

// Возврат сущности "Профиль" конкретного пользователя по переданному идентификатору
export default function useProfile(userId?: string) {
	const mainClient = useMainClient();
	return mainClient.getProfile(userId);
}
