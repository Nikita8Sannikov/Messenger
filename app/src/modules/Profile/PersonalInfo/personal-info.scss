@use "sass:math";
@use "styles/abstract/colors.scss" as *;

.info-container {
    font-weight: 400;
    &__main-info {
        border-bottom: 1px solid $light-grey;
    }
    &__username {
        font-size: 21px;
        line-height: math.div(30, 21) * 100%;
        font-weight: 600;
        text-align: start;
        margin-bottom: 10px;
    }
    &__short-info-block {
        padding: 10px 0;
        border-bottom: 1px solid $light-grey;
        font-size: 12px;
        position: relative;
        min-height: 50px;
        &:hover {
            .info-container__short-info-block__edit {
                visibility: visible;
            }
        }
        &__spinner {
            width: 100%;
            display: flex;
            justify-content: center;
            align-items: center;
        }
        &__edit {
            visibility: hidden;
            position: absolute;
            right: 0;
            top: -9px;
            background-color: $white;
            color: $medium-grey;
            padding-left: 5px;
            &:hover {
                text-decoration: underline;
                cursor: pointer;
            }
        }
        &__row {
            display: flex;
            justify-content: flex-start;
            align-items: center;
            padding: 3px 0;
            text-align: start;
            &__title {
                color: $medium-grey;
                width: 30%;
            }
            a {
                color: $blue;
                &:hover {
                    text-decoration: underline;
                }
            }
            span {
                color: $blue;
            }
        }
    }
    &__counts-block {
        display: flex;
        justify-content: space-between;
        align-items: center;
        padding: 10px 0;
        &:hover {
            cursor: pointer;
        }
        .counts-block__count {
            display: flex;
            justify-content: center;
            align-items: center;
            flex-direction: column;
            font-size: 12px;
            color: $dark-blue;
            width: 20%;
            span {
                color: $blue;
                font-weight: 400;
                font-size: 19px;
                text-align: center;
            }
        }
    }
}