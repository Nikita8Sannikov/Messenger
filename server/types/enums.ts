// API маршруты
enum ApiRoutes {
    //----auth----------
    signUp = "/sign-up",
    signIn = "/sign-in",
    logout = "/logout",
    //----userInfo------
    getMe = "/get-me",
    getUser = "/get-user",
    editInfo = "/edit-info",
    getUserDetail = "/get-user-detail",
    //----images---------
    uploadAvatar = "/upload-avatar",
    saveAvatar = "/save-avatar",
    changeAvatar = "/change-avatar",
    getPhotos = "/get-photos",
    savePhotos = "/save-photos",
    deletePhoto = "/delete-photo",
    //----file-----------
    saveFiles = "/save-files",
    openFile = "/open-file",
    downloadFile = "/download-file",
    deleteFiles = "/delete-files",
    //----friends--------
    friends = "/friends",
    getFriends = "/get-friends",
    getFriendInfo = "/get-friend-info",
    getCountFriends = "/get-count-friends",
    getPossibleUsers = "/get-possible-users",
    addToFriend = "/add-to-friend",
    unsubscribeUser = "/unsubscribe-user",
    acceptUser = "/accept-user",
    leftInSubscribers = "/left-in-subscribers",
    deleteFriend = "/delete-friend",
    getFriendsNotification = "/get-friends-notification",
    blockFriend = "/block-friend",
    checkBlockStatus = "/check-block-status",
    //----messages--------
    getDialogs = "/get-dialogs",
    deleteChat = "/delete-chat",
    getMessages = "/get-messages",
    saveMessage = "/save-message",
    updateMessage = "/update-message",
    deleteMessage = "/delete-message",
    readMessage = "/read-message",
    getChatId = "/get-chat-id",
    getMessageNotification = "/get-message-notification",
    getChatInfo = "/get-chat-info",
    getUsersInChat = "/get-users-in-chat",
    getLastSeen = "/get-last-seen",
    getChatSoundStatus = "/get-chat-sound-status",
    setChatSoundStatus = "/set-chat-sound-status",
    getAttachments = "/get-attachments",
    //------calls---------
    endCall = "/end-call",
};

// Статусы HTTP-запросов
enum HTTPStatuses {
    PermanentRedirect = 308,
    BadRequest = 400,
    Unauthorized = 401,
    Forbidden = 403,
    NotFound = 404,
    ServerError = 500,
};

// SOCKET маршруты
enum SocketActions {
    // ---------------USERS------------------
    GET_ALL_USERS = "GET_ALL_USERS",
    GET_NEW_USER = "GET_NEW_USER",
    USER_DISCONNECT = "USER_DISCONNECT",
    //----------------FRIENDS----------------
    ADD_TO_FRIENDS = "ADD_TO_FRIENDS",
    ACCEPT_FRIEND = "ACCEPT_FRIEND",
    UNSUBSCRIBE = "UNSUBSCRIBE",
    FRIENDS = "FRIENDS",
    BLOCK_FRIEND = "BLOCK_FRIEND",
    // ---------------MESSAGES---------------
    MESSAGE = "MESSAGE",
    SEND_MESSAGE = "SEND_MESSAGE",
    EDIT_MESSAGE = "EDIT_MESSAGE",
    SET_TEMP_CHAT = "SET_TEMP_CHAT",
    GET_NEW_MESSAGE_ON_SERVER = "GET_NEW_MESSAGE_ON_SERVER",
    ADD_NEW_MESSAGE = "ADD_NEW_MESSAGE",
    NOTIFY_WRITE = "NOTIFY_WRITE",
    CHANGE_READ_STATUS = "CHANGE_READ_STATUS",
    ACCEPT_CHANGE_READ_STATUS = "ACCEPT_CHANGE_READ_STATUS",
    DELETE_MESSAGE = "DELETE_MESSAGE",
    DELETE_CHAT = "DELETE_CHAT",
    // --------------CALLS-------------------
    CALL = "CALL",
    NOTIFY_CALL = "NOTIFY_CALL",
    ACCEPT_CALL = "ACCEPT_CALL",
    CHANGE_CALL_STATUS = "CHANGE_CALL_STATUS",
    SET_CALL_STATUS = "SET_CALL_STATUS",
    END_CALL = "END_CALL",
    CHANGE_STREAM = "CHANGE_STREAM",
    CANCEL_CALL = "CANCEL_CALL",
    ALREADY_IN_CALL = "ALREADY_IN_CALL",
    NOT_ALREADY_IN_CALL = "NOT_ALREADY_IN_CALL",
    IS_TALKING = "IS_TALKING",
    // --------------WEBRTC------------------
    ADD_PEER = "ADD_PEER",
    TRANSFER_CANDIDATE = "TRANSFER_CANDIDATE",
    TRANSFER_OFFER = "TRANSFER_OFFER",
    SESSION_DESCRIPTION = "SESSION_DESCRIPTION",
    GET_CANDIDATE = "GET_CANDIDATE",
    REMOVE_PEER = "REMOVE_PEER",
    //-----------------SYSTEM---------------
    SOCKET_CHANNEL_ERROR = "SOCKET_CHANNEL_ERROR",
};

// Вкладки друзей
enum FriendsTab {
    all = 0,
    online = 1,
    subscribers = 2,
    friendRequests = 3,
    incomingRequests = 4,
    search = 5,
};

// Типы сообщений
enum MessageTypes {
    MESSAGE = 1,
    WITH_FILE = 2,
    VOICE = 3,
    CALL = 4,
    FEW_FILES = 5,
};

// Типы прочитанных сообщений
enum MessageReadStatus {
    NOT_READ = 0,
    READ = 1
};

// Каналы Redis
enum RedisChannel {
    TEMP_CHAT_ID = "TEMP_CHAT_ID",
};

// Статусы звонков
enum CallStatus {
    NOT_CALL = "NOT_CALL",
    SET_CONNECTION = "SET_CONNECTION",
    WAIT = "WAIT",
    NEW_CALL = "NEW_CALL",
    ACCEPT = "ACCEPT",
    REJECTED = "REJECTED",
    OFFLINE = "OFFLINE",
};

// Разновидность звонков
enum SettingType {
    VIDEO = "VIDEO",
    AUDIO = "AUDIO"
};

// Типы ошибочных каналов сокета
enum SocketChannelErrorTypes {
    CALLS = "CALLS",
};

// Типы звонков
enum CallTypes {
    SINGLE = 0,
    GROUP = 1,
    SEPARATE = 2,
};

// Наименования звонков
enum CallNames {
    OUTGOING = "Исходящий звонок",
    INCOMING = "Входящий звонок",
    GROUP = "Групповой звонок",
    CANCEL = "Звонок отменён"
};

// Виды файлов, добавляются в объект сообщения при их обработке
enum FileVarieties {
    IMAGES = "IMAGES",
    FILES = "FILES"
};

// Типы сообщений об ошибке
enum ErrorTextsApi {
    // ошибки авторизации и аутентификации
    INCORRECT_LOGIN_OR_PASSWORD = "Не верно указан логин или пароль",
    NOT_AUTH_OR_TOKEN_EXPIRED = "Вы не авторизованы или время жизни токена сессии подошло к концу",
    YOU_ALREADY_AUTH = "Вы уже авторизированы",
    ERROR_CREATING_USER_DETAILS = "Пользователь не создался в базе данных в таблице UserDetails",
    ERROR_CREATING_USER = "Пользователь не создался в базе данных в таблице Users",
    SESSION_ID_NOT_EXISTS = "Уникальный идентификатор сессии не существует",

    // ошибки файлов, фотографий, аватаров
    SHARP_AVATAR_PATH_NOT_FOUND = "Не передан путь к сжатому файлу с аватаром",
    SHARP_PHOTO_PATH_NOT_FOUND =  "Не передан путь к сжатому файлу с фотографией",
    SHARP_PHOTO_PATHS_NOT_FOUND = "Сжатые фотографии не найдены",
    DELETE_PHOTO_PATH_NOT_FOUND = "Не передан путь для удаления фотографии",
    PHOTO_NOT_FOUND = "Фотография не найдена",
    FILES_NOT_FOUND = "Не переданы файлы",
    FILE_PATH_OPEN_NOT_FOUND = "Не передан путь для открытия файла",
    FILE_PATH_DOWNLOAD_NOT_FOUND = "Не передан путь для скачивания файла",
    FILE_NOT_FOUND = "Файл не найден",

    // ошибки в друзьях
    NOT_CORRECT_ANSER_GET_FRIENDS_NOTIFICATIONS = "Запрос выполнился некорректно и ничего не вернул (getFriendsNotification)",
    NOT_CORRECT_ANSER_POSSIBLE_USERS = "Запрос выполнился некорректно и ничего не вернул (possibleUsers)",
    NOT_CORRECT_ANSER_GET_FRIENDS_TAB_0 = "Запрос выполнился некорректно и ничего не вернул (tab=0)",
    NOT_CORRECT_ANSER_GET_FRIENDS_TAB_1 = "Список пользователей на сервере пуст (tab=1)",
    NOT_CORRECT_ANSER_GET_FRIENDS_TAB_2 = "Запрос выполнился некорректно и ничего не вернул (tab=2)",
    NOT_CORRECT_ANSER_GET_FRIENDS_TAB_3 = "Запрос выполнился некорректно и ничего не вернул (tab=3)",
    NOT_CORRECT_ANSER_GET_FRIENDS_TAB_4 = "Запрос выполнился некорректно и ничего не вернул (tab=4)",
    NOT_CORRECT_ANSER_GET_FRIENDS_TAB_5 = "Запрос выполнился некорректно и ничего не вернул (tab=5)",
    UNKNOWN_TAB = "Тип вкладки не распознан",
    NOT_TEMPORAL_CHAT_ID = "Идентификатор собеседника не найден, возможно, это временный чат",
    SUBSCRIBED_ID_NOT_FOUND = "Не передан идентификатор добавляемого пользователя",
    UNSUBSCRIBED_ID_NOT_FOUND = "Не передан идентификатор отписываемого пользователя",
    ADD_TO_FRIEND_ID_NOT_FOUND = "Не передан идентификатор принимаемого пользователя",
    LEFT_TO_SUBSCRIBED_ID_NOT_FOUND = "Не передан идентификатор подписанного пользователя",
    DELETED_ID_NOT_FOUND = "Не передан идентификатор удаляемого пользователя",
    BLOCKED_ID_NOT_FOUND = "Не передан идентификатор блокируемого пользователя",
    CHECK_BLOCKED_ID_NOT_FOUND = "Не передан id проверяемого пользователя",

    // ошибки при управлении пользователями
    USER_NOT_FOUND = "Пользователь не найден",
    USER_NOT_FOUND_IN_DATABASE = "Запись пользователя в базе данных не найдена",
    USER_ID_NOT_FOUND = "id пользователя не передано",

    // ошибки в сообщениях
    NOT_CORRECT_ANSER_GET_DIALOGS = "Запрос выполнился некорректно и ничего не вернул (getDialogs/unReadMessagesCount)",
    CHAT_ID_NOT_FOUND = "Не передан уникальный идентификатор чата",
    USER_ID_IN_MESSAGE_NOT_FOUND = "Не передан Ваш уникальный идентификатор в объекте сообщения",
    CHAT_ID_IN_MESSAGE_NOT_FOUND = "Не передан уникальный идентификатор чата в объекте сообщения",
    MESSAGE_ID_NOT_FOUND = "Не передан уникальный идентификатор сообщения",
    MESSAGE_CAN_NOT_BE_EMPTY_WITHOUT_FILES = "Ошибка при изменении сообщения, оно не может быть пустым без прикрепленных файлов.",
    CHAT_PARTNER_ID_NOT_FOUND = "Не передан уникальный идентификатор собеседника",
    CHAT_PARTNER_NOT_FOUND = "Собеседник чата не найден",
    DELETED_MESSAGE_ID_NOT_FOUND = "Не передан уникальный идентификатор удаляемого сообщения",
    YOUR_ALREADY_DELETE_THIS_MESSAGE = "Данное сообщение уже было Вами удалено",
    DELETED_CHAT_ID_NOT_FOUND = "Не передан уникальный идентификатор удаляемого приватного чата",

    // ошибки в мидлварах
    PHOTO_NOT_GIVEN = "В объекте запроса изображение не найдено",
    PHOTOS_NOT_FOUND = "Не переданы изображения для сжатия",

    // ошибка в сокет соединении
    USER_ID_NOT_GIVEN = "Не передан идентификатор пользователя",
    INCORRECT_TYPE_IN_FRIENDS = "Некорректный тип при действиях в друзьях",
    CANNOT_FIND_CALL = "Невозможно завершить звонок, так как он не найден. Возможно, он уже завершен",

    // ошибки базы данных
    ERROR_IN_CLOSE_DB = "Закрытие соединения с базой данных завершилось не удачно",
    ERROR_IN_CONNECT_DB = "Соединение с базой данных завершилось с ошибкой",

    // ошибки в Redis
    ERROR_IN_CLIENT_CONNECT = "Соединение с базой данных Redis завершилось не удачно: ",
    ERROR_IN_CLIENT_WORK = "Ошибка в работе клиента Redis: ",

    // ошибки глобальные
    START_SERVER_ERROR = "Возникла ошибка при запуске экземпляра сервера",
    UNHANDLED_SYNC_ERROR = "Не обработанная ошибка в синхронном коде",
    UNHANDLED_ASYNC_ERROR = "Не обработанная ошибка в асинхронном коде",
};

// Список ключей редиса
enum RedisKeys {
    REMEMBER_ME = "rememberMe"
};

export {
    ApiRoutes,
    HTTPStatuses,
    SocketActions,
    FriendsTab,
    MessageTypes,
    MessageReadStatus,
    RedisChannel,
    CallStatus,
    SettingType,
    SocketChannelErrorTypes,
    CallTypes,
    CallNames,
    FileVarieties,
    ErrorTextsApi,
    RedisKeys,
};