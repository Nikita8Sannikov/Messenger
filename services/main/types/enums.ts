// Каналы Redis
export enum RedisChannel {
	TEMP_CHAT_ID = "TEMP_CHAT_ID",
};

// Список ключей редиса
export enum RedisKeys {
	SESS = "sess",
	REMEMBER_ME = "rememberMe",
};

// Типы действий пользователей в разделе "Друзья"
export enum FriendActionType {
	FOLLOWING = 0,
	FRIEND = 1,
	BLOCKED = 2,
	LEFT_IN_FOLLOWERS = 3,
};