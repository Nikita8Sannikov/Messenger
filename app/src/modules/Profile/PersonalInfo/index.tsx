import React from "react";
import { useNavigate } from "react-router-dom";
import Grid from "@mui/material/Grid";
import Paper from "@mui/material/Paper";

import { useAppSelector } from "../../../hooks/useGlobalState";
import useProfile from "../../../hooks/useProfile";
import useUser from "../../../hooks/useUser";
import useUserDetails from "../../../hooks/useUserDetails";
import { selectFriendState } from "../../../state/friends/slice";
import { selectUserState } from "../../../state/user/slice";
import { FriendsTab, MainFriendTabs, Pages } from "../../../types/enums";
import { onClickBlockType } from "../Friends";
import Spinner from "../../../components/Common/Spinner";

import "./personal-info.scss";

interface IPersonalInfo {
    onClickBlock: onClickBlockType;
};

export default React.memo(function PersonalInfo({ onClickBlock }: IPersonalInfo) {
    const [loading, setLoading] = React.useState(false);

    const { photosCount } = useAppSelector(selectUserState);
    const { friendsCount, subscribersCount } = useAppSelector(selectFriendState);

    const profile = useProfile();
    const { fullName } = useUser();
    const userDetails = useUserDetails();
    const navigate = useNavigate();

    // Получаем детальную информацию о пользователе
    React.useEffect(() => {
        profile.getUserDetail(setLoading);
    }, []);

    return <Grid item>
        <Paper className="info-container paper-block">
            <div className="info-container__main-info">
                <div className="info-container__username">{fullName}</div>
            </div>

            <div className="info-container__short-info-block">
                <div className="info-container__short-info-block__edit" onClick={() => navigate(Pages.edit)}>
                    Редактировать
                </div>

                {loading
                    ? <Spinner />
                    : <>
                        <div className="info-container__short-info-block__row">
                            <div className="info-container__short-info-block__row__title">День рождения:</div>
                            <span>{userDetails.birthday}</span>
                        </div>

                        <div className="info-container__short-info-block__row">
                            <div className="info-container__short-info-block__row__title">Город:</div>
                            <span>{userDetails.city}</span>
                        </div>

                        <div className="info-container__short-info-block__row">
                            <div className="info-container__short-info-block__row__title">Место работы:</div>
                            <span>{userDetails.work}</span>
                        </div>
                    </>
                }
            </div>

            <div className="info-container__counts-block">
                <div
                    className="counts-block__count"
                    onClick={() => onClickBlock(Pages.friends, { mainTab: MainFriendTabs.allFriends })}
                >
                    <span>{friendsCount}</span> {userDetails.getFriendsText(friendsCount)}
                </div>

                <div
                    className="counts-block__count"
                    onClick={() => onClickBlock(Pages.friends, { mainTab: MainFriendTabs.allFriends, tab: FriendsTab.subscribers })}
                >
                    <span>{subscribersCount}</span> {userDetails.getSubscribersText(subscribersCount)}
                </div>

                <div className="counts-block__count" onClick={() => navigate(Pages.photos)}>
                    <span>{photosCount}</span> {userDetails.getPhotosText(photosCount)}
                </div>

                <div className="counts-block__count">
                    <span>0</span> {userDetails.getAudiosText(0)}
                </div>

                <div className="counts-block__count">
                    <span>0</span> {userDetails.getVideosText(0)}
                </div>
            </div>
        </Paper>
    </Grid>
});