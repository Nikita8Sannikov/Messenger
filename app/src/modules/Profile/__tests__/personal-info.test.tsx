import { act, fireEvent, render, screen } from "@testing-library/react";

import PersonalInfo from "@modules/profile/personal-info";
import { MainFriendTabs, Pages } from "@custom-types/enums";
import type { IUser, IUserDetails } from "@custom-types/models.types";
import useProfile from "../../../__mocks__/@hooks/useProfile";
import usePhotosStore from "../../../__mocks__/@store/photos";
import useUserStore from "../../../__mocks__/@store/user";

jest.mock("@service/i18n", () => ({
	t: (key: string) => key,
}));

jest.mock("react-router-dom", () => ({
	useNavigate: () => jest.fn(),
	useParams: () => ({ userId: "123" }),
}));

describe("PersonalInfo", () => {
	const mockProfile = {
		isMe: true,
		userService: {
			syncInfo: jest.fn(),
		},
	};

	const mockNavigate = jest.fn();

	beforeEach(() => {
		useUserStore.getState().user = {
			fullName: "Test User",
		} as IUser;
		useUserStore.getState().userDetails = {
			birthday: "1990-01-01",
			city: "Test City",
			work: "Test Work",
		} as IUserDetails;
		usePhotosStore.getState().isPhotosLoading = false;
		usePhotosStore.getState().count = 0;
		(useProfile as jest.Mock).mockReturnValue(mockProfile);
		jest.spyOn(require("react-router-dom"), "useNavigate").mockReturnValue(mockNavigate);
	});

	it("renders correctly", () => {
		const { container } = render(<PersonalInfo />);
		expect(container).toMatchSnapshot();
	});

	it("displays user information correctly", () => {
		render(<PersonalInfo />);

		expect(screen.getByText("Test User")).toBeInTheDocument();
		expect(screen.getByText("1990-01-01")).toBeInTheDocument();
		expect(screen.getByText("Test City")).toBeInTheDocument();
		expect(screen.getByText("Test Work")).toBeInTheDocument();
	});

	it("shows edit button for own profile", () => {
		render(<PersonalInfo />);
		const editButton = screen.getByText("profile-module.edit");
		expect(editButton).toBeInTheDocument();
	});

	it("does not show edit button for other user's profile", () => {
		(useProfile as jest.Mock).mockReturnValue({
			...mockProfile,
			isMe: false,
		});

		render(<PersonalInfo />);
		const editButton = screen.queryByText("profile-module.edit");
		expect(editButton).not.toBeInTheDocument();
	});

	it("shows loading spinner when photos are loading", () => {
		usePhotosStore.getState().isPhotosLoading = true;
		usePhotosStore.getState().count = 0;

		render(<PersonalInfo />);
		expect(screen.getByTestId("spinner")).toBeInTheDocument();
	});

	it("navigates to edit page when edit button is clicked", async () => {
		render(<PersonalInfo />);
		const editButton = screen.getByText("profile-module.edit");
    
		await act(async () => {
			fireEvent.click(editButton);
		});
    
		expect(mockNavigate).toHaveBeenCalledWith(Pages.edit);
	});

	it("navigates to friends page when friends count is clicked", async () => {
		render(<PersonalInfo />);
		const friendsCount = screen.getAllByText("0")[0];
    
		await act(async () => {
			fireEvent.click(friendsCount);
		});
    
		expect(mockNavigate).toHaveBeenCalledWith(Pages.friends, {
			state: { mainTab: MainFriendTabs.allFriends },
		});
	});

	it("navigates to photos page when photos count is clicked", async () => {
		render(<PersonalInfo />);
		const photosCount = screen.getAllByText("0")[2];
    
		await act(async () => {
			fireEvent.click(photosCount);
		});
    
		expect(mockNavigate).toHaveBeenCalledWith("/photos/123");
	});

	it("calls syncInfo on mount", () => {
		render(<PersonalInfo />);
		expect(mockProfile.userService.syncInfo).toHaveBeenCalled();
	});

	it("displays correct counts for all sections", () => {
		usePhotosStore.getState().count = 5;
		render(<PersonalInfo />);
    
		const counts = screen.getAllByText("0");
		expect(counts).toHaveLength(4); // Friends, Subscribers, Audios, Videos
    
		const photoCount = screen.getByText("5");
		expect(photoCount).toBeInTheDocument();
	});

	it("handles empty user details gracefully", () => {
		useUserStore.getState().userDetails = {} as IUserDetails;
		render(<PersonalInfo />);
    
		expect(screen.getByText("Test User")).toBeInTheDocument();
		expect(screen.queryByText("1990-01-01")).not.toBeInTheDocument();
		expect(screen.queryByText("Test City")).not.toBeInTheDocument();
		expect(screen.queryByText("Test Work")).not.toBeInTheDocument();
	});
});
