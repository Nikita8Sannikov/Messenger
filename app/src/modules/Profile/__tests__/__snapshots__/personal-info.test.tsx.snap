// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalInfo renders correctly 1`] = `
<div>
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row grid-item-style css-1fzlhpv-MuiGrid2-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 info-container paper-block css-1gtchvp-MuiPaper-root"
      style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="info-container__main-info"
      >
        <div
          class="info-container__username"
        >
          Test User
        </div>
      </div>
      <div
        class="info-container__short-info-block"
      >
        <div
          class="info-container__short-info-block__edit"
        >
          profile-module.edit
        </div>
        <div
          class="info-container__short-info-block__row"
        >
          <div
            class="info-container__short-info-block__row__title"
          >
            profile-module.birthday
            :
          </div>
          <span
            class="info-container__short-info-block__row__value"
          >
            1990-01-01
          </span>
        </div>
        <div
          class="info-container__short-info-block__row"
        >
          <div
            class="info-container__short-info-block__row__title"
          >
            profile-module.city
            :
          </div>
          <span
            class="info-container__short-info-block__row__value"
          >
            Test City
          </span>
        </div>
        <div
          class="info-container__short-info-block__row"
        >
          <div
            class="info-container__short-info-block__row__title"
          >
            profile-module.work
            :
          </div>
          <span
            class="info-container__short-info-block__row__value"
          >
            Test Work
          </span>
        </div>
      </div>
      <div
        class="info-container__counts-block"
      >
        <div
          class="info-container__counts-block__count"
        >
          <span
            class="info-container__counts-block__count__value"
          >
            0
          </span>
           
        </div>
        <div
          class="info-container__counts-block__count"
        >
          <span
            class="info-container__counts-block__count__value"
          >
            0
          </span>
           
        </div>
        <div
          class="info-container__counts-block__count"
        >
          <span
            class="info-container__counts-block__count__value"
          >
            0
          </span>
           
        </div>
        <div
          class="info-container__counts-block__count"
        >
          <span
            class="info-container__counts-block__count__value"
          >
            0
          </span>
           
        </div>
        <div
          class="info-container__counts-block__count"
        >
          <span
            class="info-container__counts-block__count__value"
          >
            0
          </span>
           
        </div>
      </div>
    </div>
  </div>
</div>
`;
